host-bmc/host_pdr_handler.cpp

```diff
@@ -1728,15 +1728,21 @@ void HostPDRHandler::setFRUDynamicAssociations()
         for (const auto& [rightPath, rightEntity] : objPathMap)
         {
             uint16_t rightEntityType = rightEntity.entity_type;
+
+            std::string leftPath1 = leftPath.string() + "/";
+            std::string rightPath1 = rightPath.string() + "/";
+
             // if leftpath is same as rightPath
             // then both dbus objects are same, so
             // no need to create any associations
             if (leftPath == rightPath)
             {
                 continue;
             }
-            else if ((rightPath.string().find(leftPath) != std::string::npos) ||
-                     (leftPath.string().find(rightPath) != std::string::npos))
+
+            else if ((rightPath.string().find(leftPath1) !=
```

Comment:
Can we use parent_path() to know the hierarchy?

Reply:
I did try this and it seemed to be not working, keeping the logic as izz